### PR TITLE
feat: add numberOfWords to device.read function

### PR DESCRIPTION
**Pull request by vaeng**
the write function takes the numberOfWords from the size of the passed array.

### CODE REVIEW

File: deviceaccess.py [hunk 1]
```diff
@@ -1206,16 +1206,21 @@ def getRegisterCatalogue(self) -> pb.RegisterCatalogue:
         """
         return self._device.getRegisterCatalogue()
 
-    def read(self, registerPath: str, dtype: np.dtype = np.float64, wordOffsetInRegister: int = 0,
-             accessModeFlags: Sequence[AccessMode] = None) -> np.ndarray | np.number:
+    def read(self, registerPath: str, dtype: np.dtype = np.float64, numberOfWords: int = 0, numberOfChannels: int = 0, wordOffsetInRegister: int = 0, accessModeFlags: Sequence[AccessMode] = None) -> np.ndarray | np.number:
```
Comment (mhier) [inline deviceaccess.py:1209]: isn't that line longer than 120 characters?

**Review by mhier — COMMENTED** [review]
I think there should be a test for this new feature, including exception cases like requesting a too long offset.

File: deviceaccess.py (hunk 1 above)
Comment (vaeng) [reply deviceaccess.py:1209]: It's 119.

Comment (mhier) [reply deviceaccess.py:1209]: no, it's longer, sorry. without the indentation it is 223 characters.

Comment (mhier) [reply deviceaccess.py:1209]: just to be sure: I refer to the line as it is shown by github, which claims you merged the old lines 1209 and 1210 into the single line 1209 (and added a parameter which makes it even longer). The original lines seem to be formatted properly, at least they were < 120 chars but > 80 chars.

**Pull request merged**